concept/type/impl/TypeImpl.java

```diff
@@ -22,6 +22,10 @@
 import com.vaticle.typedb.core.common.iterator.FunctionalIterator;
 import com.vaticle.typedb.core.common.parameters.Label;
 import com.vaticle.typedb.core.concept.ConceptImpl;
+import com.vaticle.typedb.core.concept.type.AttributeType;
```

Comment:
the only awkward think about this implementation is that now an `Impl` depends on a higher-level iterface that is one level higher in the package structure. However, they are still within the same bazel package

Reply:
It's actually fine because the implementation classes do tend to depend on their interfaces already, and not the other way around. In this case, we're just adding more subtypes of the interface that `TypeImpl` depends on, which is `Type.java`.